kipoiseq/extractors/vcf_matching.py

```diff
@@ -110,12 +156,37 @@ def __init__(
             pyranges object. This argument is not valid with intervals.
             Currently unused
         """
-        self.vcf = MultiSampleVCF(vcf_file, lazy=vcf_lazy)
+        self.variant_fetcher = self._read_variants(vcf_file, variants, variant_fetcher, vcf_lazy)
         self.interval_attrs = interval_attrs
         self.pr = self._read_intervals(gtf_path, bed_path, pranges,
                                        intervals, interval_attrs, duplicate_attr=True)
         self.variant_batch_size = variant_batch_size
 
+    @staticmethod
+    def _read_variants(
+            vcf_file=None,
+            variants=None,
+            variant_fetcher=None,
+            vcf_lazy: bool = True,
+    ):
+        if vcf_file is not None:
+            from kipoiseq.extractors import MultiSampleVCF
+            vcf = MultiSampleVCF(vcf_file, lazy=vcf_lazy)
+
+            if os.environ.get('PYTEST_RUNNING', '') == 'true':
```

Comment:
Can we not have this in the code but use pytest-mocker?

Reply:
Can you give an example?
How would I wrap the `MultiSampleVCF` with pytest-mocker?